### DISCUSSION TITLE
How do I compile .so (dynamic) libraries in Linux, instead of .a (static) ones?

### DISCUSSION THREAD

**Discussion by MrBolt2005** [microsoft/vcpkg#19517] [Q&A]

**Comment by srgbtl** [comment]
See: https://github.com/microsoft/vcpkg/blob/master/docs/examples/overlay-triplets-linux-dynamic.md

**Comment by PhoebeHui** [comment]
@srgbtl, thanks for your comments! @MrBolt2005, could you try above solution?

**Comment by Osyotr** [comment]
Note that built shared objects will not have rpath. This is the reason why static linkage for linux is default in vcpkg. Perhaps one day it will be fixed in vcpkg, but in meantime you may want to update your triplet file like here: https://github.com/microsoft/vcpkg/discussions/19127#discussioncomment-1049627
